packages/extension-core/src/domains/transactions/watchEthereumTransaction.ts

```diff
@@ -77,12 +77,22 @@ export const watchEthereumTransaction = async (
         )
     } catch (err) {
       log.error("watchEthereumTransaction error: ", { err })
-      updateTransactionStatus(hash, "error")
+      const error = err as Error
+      const isNotFound = error.message === "Transaction not found"
```

Comment:
Maybe this should be `error?.message` in case a string or something is thrown instead?

Seems unlikely, but if it happens then `error.message` will crash the UI.

Reply:
good catch, thanks